src/main/java/com/example/demo1/service/ShoppingCartService.java

```diff
@@ -0,0 +1,129 @@
+package com.example.demo1.service;
+
+import com.example.demo1.model.ShoppingCart;
+import com.example.demo1.model.dto.ShoppingCartDto;
+import com.example.demo1.model.dto.ShoppingCartReturnDto;
+import com.example.demo1.model.dto.ShoppingCartUpdateDto;
+import com.example.demo1.repository.ShoppingCartRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    public void create(ShoppingCartDto shoppingCartDto){
+        if (!shoppingCartDto.isAvailability())
+        {
+            throw new RuntimeException("Delivery is not available");
```

Comment:
define your own exception eg BusinessException and maybe an @ExceptionHandler too